network-interface/src/network.rs

```diff
@@ -81,6 +81,15 @@ pub trait Network: Send + Sync + Unpin + 'static {
     /// If the peer isn't found, `None` is returned.
     fn get_peer_info(&self, peer_id: Self::PeerId) -> Option<PeerInfo>;
 
+    /// Gets the set of connected peers that provide the supplied services
+    /// If we currently don't have any connected peer that provides those services,
+    /// we dial peers.
+    /// If there is no peer in the network that provides those services, we return an error
+    async fn get_peers_by_services(
+        &self,
+        services: Services,
```

Comment:
I think we're missing something like an `optimal_num_peers` that the network can use to try to return such number of peers